pkg/cluster/machine/machine.go

```diff
@@ -65,29 +66,29 @@ func ParseManifest(file string) (ml []*clusterv1.Machine, bl []*existinginfrav1.
 }
 
 // Parse parses the provided machines io.Reader.
-func Parse(r io.ReadCloser) (ml []*clusterv1.Machine, bl []*existinginfrav1.ExistingInfraMachine, err error) {
-	decoder := clusteryaml.NewYAMLDecoder(r)
-	defer decoder.Close()
-
-	for {
-		obj, _, err := decoder.Decode(nil, nil)
-		if err == io.EOF {
-			break
-		} else if err != nil {
-			return nil, nil, err
-		}
+func Parse(rc io.ReadCloser) (ml []*clusterv1.Machine, bl []*existinginfrav1.ExistingInfraMachine, err error) {
+	// Read from the ReadCloser YAML document-by-document
+	fr := serializer.NewYAMLFrameReader(rc)
```

Comment:
It's no longer necessary to close the reader?

Reply:
The `FrameReader` will close the reader when all elements have been read (when `DecodeAll` is called). Due to a libgitops bug (https://github.com/weaveworks/libgitops/issues/33) if decoding fails for some reason the reader is currently left open, that needs to be fixed over there (not a blocker for this PR).